notifier/notifier_test.go

```diff
@@ -811,3 +815,148 @@ func TestHangingNotifier(t *testing.T) {
 		})
 	}
 }
+
+// TODO: renameit and even replace TestHangingNotifier with it.
```

Comment:
WDYT?

Reply:
we can keep both of them as the current one helped fixing https://github.com/prometheus/prometheus/pull/14174/commits/5c63ef2cdc55237cff3f27175b7a148557ecc60f